fit1d.py

```diff
@@ -0,0 +1,328 @@
+
+        fitted_values = fit(x)
+
+    """
+
+    fitted = fit_once(x, y, weights, function, order)
+    if niterate <= 0:
+        return fitted
+    npts = len(x)
+    rej_weights = np.ones(npts)
+    for iter in range(niterate):
+        rms_deviation = calc_rms_deviation(x, y, weights*rej_weights, fitted)
+        deviation = (y - fitted(x))*rej_weights
+        above_hi_rej = np.where(deviation > high_reject*rms_deviation)
+        rej_weights[above_hi_rej] = 0.0
+        nhigh = len(above_hi_rej[0])
+        below_low_rej = np.where(deviation < -low_reject*rms_deviation)
+        rej_weights[below_low_rej] = 0.0
+        nlow = len(below_low_rej[0])
+        nbad = nhigh + nlow
+        if nbad == 0: break
+        fitted = fit_once(x, y, weights*rej_weights, function, order)
+    return fitted
+
+def calc_rms_deviation(x, y, weights, fitted):
+    """Calculate the weighted RMS deviation between y and fitted(x)
+
+    Parameters
+    ----------
+
+    x : 1-d ndarray
+        Input array of x-values
+
+    y : 1-d array
+        Input array of y-values
+
+    weights : 1-d array
+        Input array of weights
+
+    fitted : scipy.interpolate.fitpack2.LSQUnivariateSpline object
+        The result from running the fitting routine
+
+    Returns
+    -------
+
+    rms_deviation : float
+        The rms deviation
+
+    """
+
+    sumsq = 0.0
+    npts = len(x)
+    deviation = y - fitted(x)
+    devsquared = weights*deviation*deviation
+    sumsq = devsquared.sum()
+    sumweights = weights.sum()
+    rms_deviation = (np.sqrt(sumsq/sumweights))
+    return rms_deviation
+
+def wtrebin(x, y, weights=None, nbin=1):
+    """Like the IRAF rg_wtbin.  Bin the x, y, and weight data by averaging nbin values.
+
+    Parameters
+    ----------
+
+    x : 1-d ndarray
+        The array of independent values
+
+    y : 1-d ndarray
+        The array of dependent values.  Must have same length as x, or function
+        returns (None, None, None)
+
+    weights : 1-d ndarray or None
+        The array of weights.  If set to None, the points are given equal weight of 1.0.  If not
+        set to None, must have the same length as x and y, or the function
+        returns (None, None, None)
+
+    nbin : int
+        The number of adjacent elements of x, y, and weight that are averaged before
+        fitting is done.  If nbin=1, no averaging is done and the input arrays are returned
+
+    Returns
+    -------
+
+    (xout, yout, wout) : 3-tuple of ndarray objects
+        The binned/averaged x, y, and weight arrays
+
+    """
+
+    if nbin == 1:
+        return x, y, weights
+    nx = len(x)
+    ny = len(y)
+    if nx != ny:
+        print("Lengths of x and y vectors different, aborting")
+        return None, None, None
+    if type(weights) == type(None):
+        weights = np.ones(nx)
+    nw = len(weights)
+    if nx != nw:
+        print("Lengths of data and weight vectors different, aborting")
+        return None, None, None
```

Comment:
An exception might be more explicit here:
`raise ValueError('Lengths of data and weight vectors different')`